### PR TITLE
Feat: support addon dry-run to get the yaml results

### LINKED ISSUE #4371: support dry-run for addon to print all yamls
Addon is actually a bunch of yaml, it will help you deploy. What if we have a command that help render these yamls:

```
vela addon enable velaux --dry-run
```

And it prints all those yamls, then you can apply them?

```
vela addon enable velaux --dry-run > addon-manifest.yaml
kubectl apply -f addon-manifest.yaml
```

_Originally posted by @wonderflow in https://github.com/kubevela/kubevela/discussions/4366#discussioncomment-3134551_

### PR DESCRIPTION
**Pull request by wonderflow**
Signed-off-by: Jianbo Sun <jianbo.sjb@alibaba-inc.com>

### Description of your changes

Fixes #4371

Ref: https://github.com/kubevela/kubevela/discussions/4366

Users can use like this:

```
vela addon enable velaux --dry-run > test.yaml
kubectl apply -f test.yaml
```

or 

```
vela addon enable velaux --dry-run | kubectl apply -f -
```

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by wangyikewxgm — APPROVED** [review]

File: pkg/addon/addon.go [hunk 1]
```diff
@@ -1098,50 +1111,52 @@ func (h *Installer) dispatchAddonResource(addon *InstallPackage) error {
...
+		}
 		addOwner(o, app)
 		err = h.apply.Apply(h.ctx, o, apply.DisableUpdateAnnotation())
 		if err != nil {
 			return err
 		}
 	}
 
+	if h.dryRun {
+		fmt.Print(h.dryRunBuff.String())
```
Comment (barnettZQG) [inline pkg/addon/addon.go:1156]: It is more well to support defining the output way.

Comment (wonderflow) [reply pkg/addon/addon.go:1156]: what do you mean by defining the output way?

Comment (chivalryq) [reply pkg/addon/addon.go:1156]: `yaml` or `json`?

Comment (wonderflow) [reply pkg/addon/addon.go:1156]: Users can use like this:

```
vela addon enable velaux --dry-run > test.yaml
```

or 

```
vela addon enable velaux --dry-run | kubectl apply -f -
```

Comment (barnettZQG) [reply pkg/addon/addon.go:1156]: Will other logs become noise?

Comment (wonderflow) [reply pkg/addon/addon.go:1156]: no, I've tested and it works.

image
image

**Review by charlie0129 — APPROVED** [review]

**Pull request merged**